firmware/uimage.ksy

```diff
@@ -42,10 +80,57 @@ types:
       - id: compression_type
         type: u1
         enum: uimage_comp
-      - id: name
+        valid:
```

Comment:
Should it (and in other places too) really be validated?

Reply:
YMMV of course, but when carving (which I use it for) it allows me to quickly reject false positives, so I would recommend adding these checks.

Reply:
> YMMV of course, but when carving (which I use it for) it allows me to quickly reject false positives, so I would recommend adding these checks.

So to add a little bit more context: this mostly happens when scanning files that have the alternative magic signatures.

Reply:
`-affected-by: 435` (https://github.com/kaitai-io/kaitai_struct/issues/435#issuecomment-389597122)